### PR TITLE
Issue/138 validate prepared ml data

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

Main: 
- Add [validation class](https://github.com/openclimatefix/nowcasting_dataset/blob/issue/138-validate-prepared-ml-data/nowcasting_dataset/dataset/validate.py) which iterates over all the batches and validates them, and also find the unique days in the entire dataset. Also added [test](https://github.com/openclimatefix/nowcasting_dataset/blob/issue/138-validate-prepared-ml-data/tests/dataset/test_validate.py) for this. 
- Add [FakeDataset](https://github.com/openclimatefix/nowcasting_dataset/blob/issue/138-validate-prepared-ml-data/nowcasting_dataset/dataset/validate.py#L83) used for testing
- Add validations script

Extras:
- add 2 methods and validation to [config](https://github.com/openclimatefix/nowcasting_dataset/blob/issue/138-validate-prepared-ml-data/nowcasting_dataset/config/model.py#L59) class
- update nwp test data
- update test configs to represent test data
- extra functions for manipulating examples/[batch](https://github.com/openclimatefix/nowcasting_dataset/blob/issue/138-validate-prepared-ml-data/nowcasting_dataset/dataset/example.py#L87)

Fixes issue #138 

## How Has This Been Tested?

Unittests + extra unittest
ran the validation script for a few batches - but need to run this on remote server/gcp

- [ ] No
- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: nowcasting_dataset/consts.py [hunk 1]
```diff
@@ -55,4 +55,40 @@
...
+    "IR_108",
+    "IR_120",
+    "IR_134",
+    "VIS006",
+    "VIS008",
+    "WV_062",
+    "WV_073",
+)
+
+DEFAULT_REQUIRED_KEYS = [
```
Comment (jacobbieker) [inline nowcasting_dataset/consts.py:74]: Nice!

**Review by jacobbieker — APPROVED** [review]
LGTM! I like all the simplification done, and refactoring, looks good!

**Comment by peterdudfield** [comment]
@JackKelly gona merge this, so that [this PR](https://github.com/openclimatefix/nowcasting_dataset/pull/150) can use some of this stuff. Please do add comments

File: nowcasting_dataset/dataset/validate.py [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    ), f"Maximum GSP value is {np.nanmax(data[GSP_YIELD])} but it should be <= 1"
+    assert (
+        np.nanmin(data[GSP_YIELD]) >= 0.0
+    ), f"Maximum GSP value is {np.nanmin(data[GSP_YIELD])} but it should be >= 0"
+
+    if OBJECT_AT_CENTER in data.keys():
+        assert data[OBJECT_AT_CENTER] == "gsp"
+
+    if not batch:
+        # add an extract dimension so that its similar to batch data
```
Comment (JackKelly) [inline nowcasting_dataset/dataset/validate.py:242]: Two tiny typos (sorry for being such a pedant... and sorry if I've misunderstood!)

* extract -> extra
* its -> it's

**Review by JackKelly — COMMENTED** [review]
Looks great to me!  Awesome work!  (And sorry for taking so long to review this!)

Just one tiny comment (and it's not important!)

File: nowcasting_dataset/dataset/validate.py (hunk 2 above)
Comment (peterdudfield) [reply nowcasting_dataset/dataset/validate.py:242]: you are definitely right! my bad

**Pull request merged**